tests/test_transformation.py

```diff
@@ -0,0 +1,290 @@
+    assert exp == result
+
+
+def test_ref_no_conflict_inside_schema_pass():
+    """Test that having no conflict in schema with references passes."""
+    v1 = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+
+        "definitions": {
+            "address": {
+                "type": "object",
+                "properties": {
+                    "street_address": {"type": "string"}
+                }
+            }
+        },
+
+        "type": "object",
+
+        "properties": {
+            "billing_address": {"$ref": "#/definitions/address"},
+            "shipping_address": {
+                "allOf": [
+                    {"$ref": "#/definitions/address"},
+                    {
+                        "properties":   {
+                            "type": {"enum": ["residential", "business"]}
+                        }
+                    }
+                ]
+            }
+        }
+    }
+    exp = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+
+        "definitions": {
+            "address": {
+                "type": "object",
+                "properties": {
+                    "street_address": {"type": "string"}
+                }
+            }
+        },
+
+        "type": "object",
+
+        "properties": {
+            "billing_address": {
+                "type": "object",
+                "properties": {
+                    "street_address": {"type": "string"}
+                }
+            },
+            "shipping_address": {
+                "allOf": [
+                    {
+                        "type": "object",
+                        "properties": {
+                            "street_address": {"type": "string"}
+                        }
+                    },
+                    {
+                        "properties": {
+                            "type": {"enum": ["residential", "business"]}
+                        }
+                    }
+                ]
+            }
+        }
+    }
+    result = transform(v1)
+    assert exp == result
+
+
+def test_ref_conflict_inside_schema():
+    """Test that having an invalid reference in the schema raises
+    RefResolutionError."""
+    v1 = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+
+        "type": "object",
+
+        "properties": {
+            "billing_address": {"$ref": "#/definitions/differentaddress"},
+            "shipping_address": {
+                "allOf": [
+                    {"$ref": "#/definitions/address"},
+                    {
+                        "properties":   {
+                            "type": {"enum": ["residential", "business"]}
+                        }
+                    }
+                ]
+            }
+        }
+    }
+    with pytest.raises(jsonschema.exceptions.RefResolutionError):
+        transform(v1)
+
+
+def test_ref_conflict_between_schemas():
+    """Test that having a conflict between schemas with references passes."""
```

Comment:
Can you explain what you mean by "conflict"? Is it two references with JSON-pointers which are similar after the `#`?